price-estimator/src/filter.rs

```diff
@@ -127,20 +168,43 @@ async fn get_markets<T>(
     Ok(warp::reply::json(&result))
 }
 
+async fn get_token_info(
+    token_id: u16,
+    token_info_fetching: &dyn TokenInfoFetching,
+) -> Result<TokenBaseInfo, Rejection> {
+    token_info_fetching
+        .get_token_info(TokenId(token_id))
+        .await
+        .map_err(|_| reject::custom(NoTokenInfo))
+}
+
 async fn estimate_buy_amount<T>(
     token_pair: TokenPair,
-    sell_amount_in_quote: u128,
-    _query: QueryParameters,
+    sell_amount_in_quote: f64,
+    query: QueryParameters,
     price_rounding_buffer: f64,
     orderbook: Arc<Orderbook<T>>,
-) -> Result<impl warp::Reply, Infallible> {
+    token_infos: Arc<dyn TokenInfoFetching>,
+) -> Result<impl Reply, Rejection> {
+    let sell_amount_in_quote_atoms = if query.atoms {
+        sell_amount_in_quote
+    } else {
+        let token_info = get_token_info(token_pair.sell_token_id, token_infos.as_ref()).await?;
+        sell_amount_in_quote * token_info.unit_in_atoms()
+    };
```

Comment:
I was wondering if this logic should rather live in the estimate_buy_amount_filter (so that the actual handler can only deal with atom based logic) but that doesn't nicely work with the way that results are returned at the moment since they also have to be changed if `atoms = false`. 
Given that the handler itself just delegates a call into the pricegraph, maybe it's fine to keep this logic here.

Reply:
Maybe it can be implemented in the `filter` by using a `map` function?

Reply:
I don't think this will turn out better by doing at a higher level so I'll keep it this way. I'm not opposed to that in the future either just don't think it's worth trying for this PR.